packages/netlify-cms-core/src/backend.ts

```diff
@@ -492,8 +492,9 @@ export class Backend {
   }
 
   async deleteLocalDraftBackup(collection: Collection, slug: string) {
-    const key = getEntryBackupKey(collection.get('name'), slug);
-    await localForage.removeItem(key);
+    await localForage.removeItem(getEntryBackupKey(collection.get('name'), slug));
+    // delete new entry backup if not deleted
+    slug && (await localForage.removeItem(getEntryBackupKey(collection.get('name'))));
```

Comment:
> 3. See that the new entry backup it deleted

Yes, it would, after step 1, using the editor toolbar back button that displays a popup fire the delete backup call.

Reply:
In the scenario I described I didn't click the back button and wasn't shown any pop up.